tests/meta/demultiplex/test_demux_post_processing.py

```diff
@@ -482,3 +482,38 @@ def test_finish_all_flowcells(
 
     # THEN we should log that we are checking flow cell
     assert f"Check demultiplexed flow cell {bcl2fastq_flow_cell.full_name}" in caplog.text
+
+
+def test_is_bcl2fastq_folder_structure(
+    demultiplex_context: CGConfig, bcl2fastq_folder_structure: Path
+):
+    """Test is_bcl2fastq_demux_folder_structure with a folder structure that follows the bcl2fastq folder structure."""
+    # GIVEN a bcl2fastq folder structure
+    mock_demux_post_processing_api = DemuxPostProcessingAPI(demultiplex_context)
+    mock_demux_post_processing_api.flow_cell_dir = bcl2fastq_folder_structure
```

Comment:
Is this actually a mock? You're instantiating the real object just setting the flow_cell_dir. I think it can just be called `demux_post_processing_api`

Reply:
Yes, you are right,. I intended to use a mock which was not necessary in the end!